explainaboard/metrics/accuracy.py

```diff
@@ -17,6 +19,19 @@
 from explainaboard.utils.typing_utils import narrow
 
 
+@dataclass
+class ConfidenceMetricResult(AuxiliaryMetricResult):
+    """The result of the confidence metric for calibration analysis.
+
+    This is an auxiliary result of Accuracy metric result.
+
+    Args:
+        confidence: The average confidence of a bucket of sampels.
+    """
+
+    confidence: float
+
+
```

Comment:
There is already `Score` to store any float values. We basically don't need to implement `AuxiliaryMetricResult` anymore. You instead need to add additional results with `Score` in `evaluate_from_stats`.

https://github.com/neulab/ExplainaBoard/blob/00f1a6db749e6dd9b171d7192cbc7c0afa70fb75/explainaboard/metrics/metric.py#L43

Reply:
Defining ConfidenceMetricResult inheriting AuxiliaryMetricResult is because the Performance class is defined as 
```
class Performance:
    value: float
    confidence_score_low: float | None = None
    confidence_score_high: float | None = None
    auxiliary_result: AuxiliaryMetricResult | None = None
```
If we add confidence value to the `MetricResult` in `evaluate_from_stats`, we cannot see it in `Performance` and still need to create a `AuxiliaryMetricResult`.

Reply:
I see. We need the following approach:

- Keep `ConfidenceMetricResult` for now.
- `metric.evaluate_from_stats` should return `Score` with name "confidence" because AuxiliaryMetricResult is being deprecated. 
- Repack the "confidence" value in `Score` to `ConfidenceMetricResult` when `MetricResult` is converted to `Performance`

This is the same situation with "agreement" in `ExternalEval`, which is currently implemented as `Score`.
https://github.com/neulab/ExplainaBoard/blob/9ba998c125587bb5a9b421706b1ca50028af220b/explainaboard/metrics/external_eval.py#L175